### PR TITLE
Add `fix-lock-files` command + other improvements

### PR DESCRIPTION
**Pull request by bonsairobo**
I started out wanting to add a new command that would check + fix up `Cargo.lock` files as described in [this RFD](https://github.com/ForesightMiningSoftwareCorporation/wiki/pull/99). This led me to do a bit of refactoring and add some testing to existing logic as well.

I've done a bit of work on the CLI, trying to extract some of the core logic into smaller methods. I ended up with a `CrateGraph` object that does all of the workspace discovery and change detection. Some of the logic that existed in `check_workspace` has been replaced. Care should be taken reviewing that revision to be sure I didn't break anything. I'm not sure how we should go about testing this end-to-end.

I've run the new command `fslabscli fix-lock-files` on `fsl_libs` and it seems to work as expected. Since we've started tracking lock files, I've noticed some are already out of date, so I generated [this PR](https://github.com/ForesightMiningSoftwareCorporation/fsl_libs/pull/1445) using the new command.

### CODE REVIEW

**Review by loispostula — COMMENTED** [review]
Thank you for this.

My only comment is regarding the location of the actual `fix_lock_files` method.

I understand it make sense to have it in `crate_graph.rs` but the other commands do their stuff in their command root, so it's a bit inconsistent now.

What do you think?

**Comment by bonsairobo** [comment]
@loispostula 

I think there is a benefit to having separate layers of abstraction for the CLI commands and the methods that implement the core logic. The CLI command modules should mostly be concerned with translating CLI arguments into the more object-oriented API. Then the objects can be easily unit-tested without thinking about the CLI.

I wouldn't worry too much about that inconsistency I guess. But I'm open to thoughts on other ways to organize the `crate_graph` module. It could potentially be broken up further.

File: src/crate_graph.rs [hunk 1]
```diff
@@ -138,6 +146,84 @@ impl From<&Package> for PackageKey {
...
+                        }
+                    }
+                }
+            }
+        }
+
+        me
+    }
+
+    /// Given a set `seed` of **relative** paths to packages into the repo,
```
Comment (bonsairobo) [inline src/crate_graph.rs:324]: "into" --> "in"

File: src/crate_graph.rs [hunk 2]
```diff
@@ -138,6 +146,84 @@ impl From<&Package> for PackageKey {
...
+    /// returns the superset of packages that directly or indirectly depend on
+    /// one of the packages in `seed`.
```
Comment (bonsairobo) [inline src/crate_graph.rs:326]: "one" --> "any"

**Review by loispostula — APPROVED** [review]

**Pull request merged**